hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/TestDeleteContainerHandler.java

```diff
@@ -301,26 +319,13 @@ public void testDeleteNonEmptyContainerBlockTable()
     Assert.assertFalse(isContainerDeleted(hddsDatanodeService,
         containerId.getId()));
 
-    // Set container blockCount to 0 to mock that it is empty as per RocksDB
-    getContainerfromDN(hddsDatanodeService, containerId.getId())
-        .getContainerData().setBlockCount(0);
-    // Write entries to the block Table.
-    try (DBHandle dbHandle
-             = BlockUtils.getDB(
-                 (KeyValueContainerData)getContainerfromDN(hddsDatanodeService,
-                     containerId.getId()).getContainerData(),
-        conf)) {
-      BlockData blockData = new BlockData(new BlockID(1, 1));
-      dbHandle.getStore().getBlockDataTable().put("block1", blockData);
-    }
-
-    long containerDeleteFailedNonEmptyBefore =
-        metrics.getContainerDeleteFailedNonEmptyDir();
```

Comment:
Since we have already created key before and not deleted, Key will be in block table. So again writing into rocks db block table may not be required. We are just deleting blocks from the disk here. In this case block table has entry but disk don't have, So in this case container will not be empty and delete is not allowed unless we do force delete.